test/unit/spec/common/jobManager-Spec.js

```diff
@@ -964,37 +1051,78 @@ define([
                     spyOn(console, 'error');
                     spyOn(this.jobManagerInstance, 'updateModel').and.callThrough();
                     // the first message will not trigger `updateModel` as it is identical to the existing jobState
-                    this.jobManagerInstance.runHandler(jcm.MESSAGE_TYPE.STATUS, messageOne, jobId);
-                    expect(this.jobManagerInstance.model.getItem(`exec.jobState`)).toEqual(
-                        jobState
-                    );
-                    expect(console.error).toHaveBeenCalledTimes(1);
-                    expect(this.jobManagerInstance.updateModel).toHaveBeenCalledTimes(0);
-
-                    // the second message will trigger `updateModel`
-                    this.jobManagerInstance.runHandler(jcm.MESSAGE_TYPE.STATUS, messageTwo, jobId);
-                    expect(this.jobManagerInstance.model.getItem(`exec.jobState`)).toEqual(
-                        updateOne
-                    );
-                    expect(console.error).toHaveBeenCalledTimes(2);
-                    expect(this.jobManagerInstance.updateModel).toHaveBeenCalledTimes(1);
-                    let updateModelCallArgs = this.jobManagerInstance.updateModel.calls.allArgs();
-                    // updateModel takes [jobState] as an argument
-                    expect(updateModelCallArgs).toEqual([[[updateOne]]]);
-
-                    // third message will also trigger an update
-                    this.jobManagerInstance.runHandler(
-                        jcm.MESSAGE_TYPE.STATUS,
-                        messageThree,
-                        jobId
-                    );
-                    expect(this.jobManagerInstance.model.getItem(`exec.jobState`)).toEqual(
-                        updateTwo
-                    );
-                    expect(console.error).toHaveBeenCalledTimes(3);
-                    expect(this.jobManagerInstance.updateModel).toHaveBeenCalledTimes(2);
-                    updateModelCallArgs = this.jobManagerInstance.updateModel.calls.allArgs();
-                    expect(updateModelCallArgs).toEqual([[[updateOne]], [[updateTwo]]]);
+                    return new Promise((resolve) => {
```

Comment:
Prepare for some hideousness here as we enter nesting hell...

Reply:
The operation of this test has not changed apart from having the message come in over the bus, rather than triggering `runHandler` directly.